### PR TITLE
SPV: Change barrier emission to conform to Khronos directions

### PR DESCRIPTION
**Pull request by johnkslang**
The memory model group agreed to these directions for how to map GLSL barrier, memoryBarrier, etc. Corresponding HLSL built-ins follow suit.

~Non-matching HLSL built-ins are still TBD.~

### CODE REVIEW

File: SPIRV/GlslangToSpv.cpp [hunk 1]
```diff
@@ -5431,29 +5431,47 @@ spv::Id TGlslangToSpvTraverser::createNoArgOperation(glslang::TOperator op, spv:
...
-        builder.createMemoryBarrier(spv::ScopeDevice, spv::MemorySemanticsCrossWorkgroupMemoryMask);
+        builder.createMemoryBarrier(spv::ScopeWorkgroup, spv::MemorySemanticsAllMemory |
+                                                         spv::MemorySemanticsAcquireReleaseMask);
         return 0;
     case glslang::EOpAllMemoryBarrierWithGroupSync:
         // Control barrier with non-"None" semantic is also a memory barrier.
-        builder.createControlBarrier(spv::ScopeDevice, spv::ScopeDevice, spv::MemorySemanticsAllMemory);
+        builder.createControlBarrier(spv::ScopeDevice, spv::ScopeDevice,
+                                        spv::MemorySemanticsAllMemory |
+                                        spv::MemorySemanticsSequentiallyConsistentMask);
```
Comment (jeffbolznv) [inline SPIRV/GlslangToSpv.cpp]: This one is not quite right, and I think you meant to make the HLSL changes in a separate commit?

Comment (johnkslang) [reply SPIRV/GlslangToSpv.cpp]: Yes, this is a non-functional change (SequentiallyConsistent was inside AllMemory before). All the HLSL-specific changes will be in an upcoming commit.)

File: SPIRV/GlslangToSpv.cpp [hunk 2]
```diff
@@ -5431,40 +5432,65 @@ spv::Id TGlslangToSpvTraverser::createNoArgOperation(glslang::TOperator op, spv:
...
     case glslang::EOpGroupMemoryBarrier:
-        builder.createMemoryBarrier(spv::ScopeDevice, spv::MemorySemanticsCrossWorkgroupMemoryMask);
+        builder.createMemoryBarrier(spv::ScopeWorkgroup, spv::MemorySemanticsAllMemory |
+                                                         spv::MemorySemanticsAcquireReleaseMask);
         return 0;
     case glslang::EOpAllMemoryBarrierWithGroupSync:
-        // Control barrier with non-"None" semantic is also a memory barrier.
-        builder.createControlBarrier(spv::ScopeDevice, spv::ScopeDevice, spv::MemorySemanticsAllMemory);
+        builder.createControlBarrier(spv::ScopeWorkgroup, spv::ScopeDevice,
+                                        spv::MemorySemanticsAllMemory |
```
Comment (antiagainst) [inline SPIRV/GlslangToSpv.cpp]: The memory semantics for this should be Image|Uniform|Workgroup|AcquireRelease?

Comment (johnkslang) [reply SPIRV/GlslangToSpv.cpp]: Are you asking about saying Image|Uniform|Workgroup versus MemorySemanticsAllMemory?

That is a common theme. Need to discuss with the memory model group.

Comment (johnkslang) [reply SPIRV/GlslangToSpv.cpp]: Will prune this back to Image|Uniform|Workgroup|Atomic.
